### PR TITLE
fix: export Typescript definitions and fix resulting type errors

### PR DESCRIPTION
**Pull request by alexmcmanus**
Few of the modules export their Typescript definitions, despite being written in Typescript. This
makes them more difficult to use standalone from Typescript code. As a result of exporting the
definitions, some minor type errors emerge.

#2451

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-util-mock/src/CFNParser/appsync-resource-processor.ts [hunk 1]
```diff
@@ -257,16 +259,21 @@ export function processResources(
...
+    },
     resolvers: [],
     functions: [],
     dataSources: [],
     mappingTemplates: [],
     tables: [],
     appSync: {
       name: '',
-      defaultAuthenticationType: {},
+      defaultAuthenticationType: {
```
Comment (yuth) [inline packages/amplify-util-mock/src/CFNParser/appsync-resource-processor.ts]: Why OIDC as default auth type? I would prefer this to be APIKey or cognito

File: packages/amplify-util-mock/src/__e2e__/versioned-model-transformer.e2e.test.ts [hunk 2]
```diff
@@ -29,14 +29,7 @@ beforeAll(async () => {
...
         new DynamoDBModelTransformer(),
-        new ModelAuthTransformer({
-          authConfig: {
-            defaultAuthentication: {
-              authenticationType: 'API_KEY',
-            },
-            additionalAuthenticationProviders: [],
-          },
-        }),
+        new ModelAuthTransformer(),
```
Comment (yuth) [inline packages/amplify-util-mock/src/__e2e__/versioned-model-transformer.e2e.test.ts]: why was this removed?

File: packages/amplify-util-mock/src/CFNParser/appsync-resource-processor.ts (hunk 1 above)
Comment (alexmcmanus) [reply packages/amplify-util-mock/src/CFNParser/appsync-resource-processor.ts]: Changed to API_KEY.

File: packages/amplify-util-mock/src/__e2e__/versioned-model-transformer.e2e.test.ts (hunk 2 above)
Comment (alexmcmanus) [reply packages/amplify-util-mock/src/__e2e__/versioned-model-transformer.e2e.test.ts]: This is an interesting one. Following back to the import, you see:
```ts
import {
  default as ModelAuthTransformer,
  default as VersionedModelTransformer,
} from 'graphql-versioned-transformer';
```
So, `ModelAuthTransformer` is in fact `VersionedModelTransformer` and the TS parser correctly rejects the `authConfig` parameters. I've no idea whether it makes sense to have a `VersionedModelTransformer` masquerading as a `ModelAuthTransformer`, so otherwise I left it alone. 

This does highlight why it's useful to get the type checking working across packages.

Comment (alexmcmanus) [reply packages/amplify-util-mock/src/__e2e__/versioned-model-transformer.e2e.test.ts]: If I make that change, there will be no import of `ModelAuthTransformer`. Just so I'm clear, are you asking me to drop `ModelAuthTransformer` from the list of transformers?

Comment (attilah) [reply packages/amplify-util-mock/src/__e2e__/versioned-model-transformer.e2e.test.ts]: That import statement imports whatever is the default with an alias, basically importing the VersionTransformer class with 2 names, which is not good, but did not make harm. Since in the e2e test no ```@auth``` is used I think we don't need the 'real' ModelAuthTransformer imported.

Comment (alexmcmanus) [reply packages/amplify-util-mock/src/__e2e__/versioned-model-transformer.e2e.test.ts]: Done.
